### PR TITLE
Add ReasonML lexer.

### PR DESCRIPTION
**Pull request by fhammerschmidt**
This implements #1355 

I derived the ReasonML lexer from the OCaml one, since the former language is based on the latter, which turned out relatively good.

I added a testfile "test.re", too if you want to see for yourself.

Thanks in advance!

### CODE REVIEW

File: pygments/lexers/ml.py [hunk 1]
```diff
@@ -767,3 +766,94 @@ class OpaLexer(RegexLexer):
             (r'[^\-]+|-', Comment),
         ],
     }
+
+class ReasonLexer(RegexLexer):
+    """
+    For the ReasonML language.
+
+    .. versionadded:: 2.5
```
Comment (Anteru) [inline pygments/lexers/ml.py]: Please change to 2.6, the 2.5 ship has sailed.

File: pygments/lexers/ml.py [hunk 2]
```diff
@@ -767,3 +766,94 @@ class OpaLexer(RegexLexer):
             (r'[^\-]+|-', Comment),
         ],
     }
+
+class ReasonLexer(RegexLexer):
+    """
+    For the ReasonML language.
```
Comment (Anteru) [inline pygments/lexers/ml.py]: Could you please add a link to the language here?

**Comment by jordwalke** [comment]
Thanks @fhammerschmidt !

**Comment by montogeek** [comment]
@fhammerschmidt Do you have a plan to do the same for ReScript?

**Comment by fhammerschmidt** [comment]
> @fhammerschmidt Do you have a plan to do the same for ReScript?
Yeah, I could do that. 

But it may be wiser to do it when more of the syntax has settled (like local opens, Infix, module imports).

**Pull request merged**